src/main/java/com/ionosenterprise/sdk/S3Key.java

```diff
@@ -0,0 +1,83 @@
+package com.ionosenterprise.sdk;
+
+import com.ionosenterprise.rest.client.RestClientException;
+import com.ionosenterprise.rest.domain.S3Keys;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+
+public class S3Key extends BaseAPI {
+
+    public S3Key() throws Exception {
+        super("s3keys", "um/users");
+    }
+
+    /**
+     * Retrieve the list of S3Keys for an user.
+     *
+     * @param userId
+     * @return
+     */
+    public S3Keys getAllS3Keys(String userId) throws RestClientException, IOException {
+        return client.get(getUrlBase().concat(parentResource).concat("/").concat(userId).concat("/").concat(resource)
```

Comment:
In general I personally consider this a bad practice for several reasons:
- Text/URL readability, hard to determine which is the actual endpoint 
- No format validation
- No URL encoding in case you need one
- You add boiler plate '.concat' each time you add a sepparator "/"

My advice would be to use UriBuilder w/o UriTemplate 

Same goes for all places.

Reply:
I followed the existing already implementation but this is a very good aspect to improve. Because there are many changes required for the all API classes I've created a Jira task (VDC-5296) under which this improvement will be done.